code/modules/surgery/organs/heart.dm

```diff
@@ -254,3 +254,192 @@
+
+
+///Actually spawns the crystal which puts the ethereal in it.
+/obj/item/organ/heart/ethereal/proc/crystalize(mob/living/ethereal)
+	if(!COOLDOWN_FINISHED(src, crystalize_cooldown) || ethereal.stat != DEAD)
+		return //Should probably not happen, but lets be safe.
+	COOLDOWN_START(src, crystalize_cooldown, INFINITY) //Prevent cheeky double-healing until we get out, this is against stupid admemery
+	current_crystal = new(get_turf(ethereal), src)
+	stop_crystalization_process(ethereal)
+
+///Stop the crystalization process, unregistering any signals and resetting any variables.
+/obj/item/organ/heart/ethereal/proc/stop_crystalization_process(mob/living/ethereal)
+	UnregisterSignal(ethereal, COMSIG_HUMAN_DISARM_HIT)
+	UnregisterSignal(ethereal, COMSIG_PARENT_EXAMINE)
+	UnregisterSignal(ethereal, COMSIG_MOB_APPLY_DAMGE)
+
+	crystalization_process_damage = 0 //Reset damage taken during crystalization
+
+	if(crystalize_timer_id)
+		deltimer(crystalize_timer_id)
+		crystalize_timer_id = null
+
+
+///Lets you stop the process with enough brute damage
+/obj/item/organ/heart/ethereal/proc/on_take_damage(datum/source, damage, damagetype, def_zone)
+	SIGNAL_HANDLER
+	if(damagetype != BRUTE)
+		return
+
+	crystalization_process_damage += damage
+
+	if(crystalization_process_damage < BRUTE_DAMAGE_REQUIRED_TO_STOP_CRYSTALIZATION)
+		return
+
+	var/mob/living/carbon/human/ethereal = source
+
+	to_chat(ethereal, "<span class='userwarning'>The crystals on your body have completely broken</span>")
+	ethereal.visible_message("<span class='notice'>The crystals on [ethereal] are completely shattered and stopped growing</span>")
+
+	stop_crystalization_process(ethereal)
+
+/obj/structure/ethereal_crystal
+	name = "Ethereal Ressurection Crystal"
+	desc = "It seems to contain the corpse of an Ethereal mending its wounds."
+	icon = 'icons/obj/ethereal_crystal.dmi'
+	icon_state = "ethereal_crystal"
+	damage_deflection = 0
+	max_integrity = 100
+	resistance_flags = FIRE_PROOF
+	density = TRUE
+	anchored = TRUE
+	///The organ this crystal belongs to
+	var/obj/item/organ/heart/ethereal/ethereal_heart
+	///Timer for the healing process. Stops if destroyed.
+	var/crystal_heal_timer
+
+/obj/structure/ethereal_crystal/Initialize(mapload, obj/item/organ/heart/ethereal/ethereal_heart)
+	. = ..()
+	src.ethereal_heart = ethereal_heart
+	ethereal_heart.owner.visible_message("<span class='notice'>The crystals fully encase [ethereal_heart.owner]</span>")
+	to_chat(ethereal_heart.owner, "<span class='notice'>You are encased in a huge crystal.</span>")
+	playsound(get_turf(src), 'sound/effects/ethereal_crystalization.ogg', 50)
+	ethereal_heart.owner.forceMove(src) //put that ethereal in
+	add_atom_colour(ethereal_heart.ethereal_color, FIXED_COLOUR_PRIORITY)
+	crystal_heal_timer = addtimer(CALLBACK(src, .proc/heal_ethereal), CRYSTALIZE_HEAL_TIME, TIMER_STOPPABLE)
+	set_light(4, 10, ethereal_heart.ethereal_color)
+	update_icon()
```

Comment:
might want to add a signal check for the heart being removed from the ethereal inside of this thing somehow. for example, a wizard could cast instant summons to recall it to their hand.

the same thing applies for a signal for the heart or the ethereal inside of the crystal being destroyed somehow

Reply:
already handled